test/unit/node/settings.test.ts

```diff
@@ -0,0 +1,65 @@
+import { logger } from "@coder/logger"
+import { promises as fs } from "fs"
+import { SettingsProvider, CoderSettings } from "../../../src/node/settings"
+import { clean, mockLogger, tmpdir } from "../../utils/helpers"
+
+describe("settings", () => {
+  describe("with invalid JSON in settings file", () => {
+    const testName = "settingsInvalidJSON"
+    let testDir = ""
+    let mockSettingsFile = "coder.json"
+    let pathToMockSettingsFile = ""
+
+    beforeEach(async () => {
+      mockLogger()
+      await clean(testName)
+      testDir = await tmpdir(testName)
+      pathToMockSettingsFile = `${testDir}/${mockSettingsFile}`
+      // Missing a quote, which makes it invalid intentionally
+      await fs.writeFile(pathToMockSettingsFile, '{"fakeKey":true,"helloWorld:"test"}')
+    })
+    afterEach(async () => {
+      await fs.rm(testDir, { recursive: true, force: true })
```

Comment:
No need, this will get cleaned up on the next test run.

Leaving the directory is nice because you can inspect it if something failed in the test.

Reply:
Oh! With `clean()`? Got it!